langchain4j-core/src/main/java/dev/langchain4j/model/chat/ChatLanguageModel.java

```diff
@@ -82,17 +141,8 @@ default Response<AiMessage> generate(List<ChatMessage> messages, List<ToolSpecif
      * @return The response generated by the model.
      * {@link AiMessage} contains a request to execute the specified tool.
      */
+    @Deprecated(forRemoval = true) // TODO
```

Comment:
Do not deprecate this (and in streaming model) method before `ChatRequest` supports `ToolMode`

Reply:
`ToolChoice` is supported now.
Since `chat()` API supports all features of the old `generate()` API, we can safely deprecate it.